### PR TITLE
ADBDEV-4904-26: Remove the isnull check for NULL.

### PR DESCRIPTION
**Pull request by red1452**
Remove the isnull check for NULL.

The slot_getsysattr function is always called with the isnull argument not equal
to NULL, so I changed the isnull comparison with NULL to an assertion and also
added an assertion to the heap_getsysattr function.

### CODE REVIEW

**Comment by RekGRpth** [comment]
I think you should replace the `isnull` check with NULL in the `slot_getsysattr` function with an assertion because it is never called with a null last argument.

**Comment by RekGRpth** [comment]
add title to PR description

**Comment by red1452** [comment]
> I think you should replace the `isnull` check with NULL in the `slot_getsysattr` function with an assertion because it is never called with a null last argument.

The heap_getsysattr function is called not only from the slot_getsysattr function. At the slot_getsysattr there is check (`isnull` id not NULL) before using.

**Comment by RekGRpth** [comment]
> At the slot_getsysattr there is check (`isnull` id not NULL) before using.

So I propose replacing this condition with an assertion.

**Comment by red1452** [comment]
> add title to PR description

done.

**Comment by RekGRpth** [comment]
> > At the slot_getsysattr there is check (`isnull` id not NULL) before using.
> 
> So I propose replacing this condition with an assertion.

I suggest replace https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/execTuples.c#L1547-L1548 with
```c
        Asssert(isnull != NULL);
        *isnull = false;
```

File: src/backend/access/common/heaptuple.c [hunk 1]
```diff
@@ -556,6 +556,7 @@ heap_getsysattr(HeapTuple tup, int attnum, TupleDesc tupleDesc, bool *isnull)
 
 	Assert(tup);
 	Assert(!is_memtuple((GenericTuple) tup));
+	Assert(isnull);
```
Comment (bandetto) [inline src/backend/access/common/heaptuple.c]:
```suggestion
	Assert(isnull != NULL);
```
To be consistent

Comment (RekGRpth) [reply src/backend/access/common/heaptuple.c]: applied

**Review by bandetto — APPROVED** [review]

**Pull request merged**